### PR TITLE
[Domain Purchasing during Site Creation] Add tracks events

### LINKED ISSUE #17915: Add tracks events
## Description

Add analytics events for the domain purchasing during site creation flow.

#### Subtasks (events)

- [ ] `enhanced_site_creation_domains_selected`
   - Add new property: `domain_cost`: `free`/`domain.cost` 
- [ ] `domains_purchase_webview_viewed` → when the domain checkout webview is shown
   - Add a new `origin` property: `site_creation` / `menu`   
- [ ] `domains_registration_form_submitted` → when the form is submitted in the webview
- [ ] `domains_purchase_domain_success` → when the purchase was successful
- [ ] Site created with paid domain event

**Note:** All development should be behind [the feature flag](https://github.com/wordpress-mobile/WordPress-iOS/issues/20073)

### PR DESCRIPTION
**Pull request by antonis**
Fixes #17915

## Description
1. `enhanced_site_creation_domains_selected`
   - Adds new property `domain_cost`: `free`/`domain.cost` with https://github.com/wordpress-mobile/WordPress-Android/pull/17967/commits/59ee6c52aed15dabf8d75c8765c0f35f7b297d7b
2. `domains_purchase_webview_viewed` → when the domain checkout webview is shown
   - Adds a new `origin` property: `site_creation` / `menu` https://github.com/wordpress-mobile/WordPress-Android/pull/17967/commits/f7fcccd2510fb58b93a48cfe5cfd40e8464a4fa0 
3. `domains_registration_form_submitted` → when the form is submitted in the webview
   - Moves the tracking function in a helper class so that it can be reused if needed https://github.com/wordpress-mobile/WordPress-Android/pull/17967/commits/f7fcccd2510fb58b93a48cfe5cfd40e8464a4fa0
4. `domains_purchase_domain_success` → when the purchase was successful
   - Moves the tracking function in a helper class so that it can be reused if needed https://github.com/wordpress-mobile/WordPress-Android/pull/17967/commits/f7fcccd2510fb58b93a48cfe5cfd40e8464a4fa0
5. Site created with paid domain event
   - This could tracked in (1) but we should re-examine if a new property is needed in the [site_created event](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt#L114)

## To test:
1. Launch & Login Jetpack
2. Enable `Site Creation Domain Purchasing` feature flag

### Domains Selected
3. Open "My Sites".
4. Tap ➕ icon to create a new site
5. Tap "Create Wordpress.com site"
6. Finish the flow and tap "Done".
7. Verify if the `enhanced_site_creation_domains_selected` event has been fired with the new `domain_cost` parameter.
8. Repeat the flow with Feature Flag disabled and verify new parameter is not in the event.

### Webview Viewed
3. Enable feature flag
4. Menu -> Domains -> "Search for a domain" -> Select one option and tap "Select domain"
5. Verify if the `domains_purchase_webview_viewed` event has been fired with the new `origin` parameter. The parameter must be "menu". The other option will be available after that screen is implemented.
6. Repeat the steps with the feature flag turned off and verify the `origin` parameter is not in the event.

I did not add the other 2 events mentioned in the issue as I am not sure how to test them also `domains_registration_form_submitted` seem to already exist in code. I think we need revisit them once the flow is in place.

## Regression Notes
1. Potential unintended areas of impact
Existing domain registration function

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and existing tests

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/domains/DomainsRegistrationTracker.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package org.wordpress.android.ui.domains
+
+import org.wordpress.android.analytics.AnalyticsTracker
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class DomainsRegistrationTracker @Inject constructor(val tracker: AnalyticsTrackerWrapper) {
```
Comment (ovitrif) [inline WordPress/src/main/java/org/wordpress/android/ui/domains/DomainsRegistrationTracker.kt]: Thank you for extracting the domains tracking to this dedicated `tracker` 🙇🏻

**Review by ovitrif — CHANGES_REQUESTED** [review]
Great work! Progress LGTM up to this point @antonis 🙇🏻 

I didn't see anything logic that depends on the feature flag in the changes though 🤷🏻 , which explains why during testing the new event properties are tracked regardless of the feature flag state (ON/OFF).

## Testing:
- ### 🌕 Domains Selected - domain_cost property
  - #### 🟢 Feature flag ON
    ```
    17:56:40.531 WordPress-STATS      com.jetpack.android.beta   I  🔵 Tracked: enhanced_site_creation_domains_selected, Properties: {"chosen_domain":"testedbyovitracksdomains001.news.blog","search_term":"testedbyovitracksdomains001","domain_cost":"free"}
    ```
  - #### 🔴 Feature flag OFF
    The event was still fired with the `domain_cost` property for me:

    ```
    17:57:34.882 WordPress-STATS      com.jetpack.android.beta   I  🔵 Tracked: enhanced_site_creation_domains_selected, Properties: {"chosen_domain":"testedbyoviwopurchasingfeaturetracks002.wordpress.com","search_term":"testedbyoviwopurchasingfeaturetracks002","domain_cost":"free"}
    ```

  The experiment event was not fired though, it's only once in my logs:
    ```
    17:55:38.759 WordPress-STATS      com.jetpack.android.beta   I  🔵 Tracked: enhanced_site_creation_domain_purchasing_experiment, Properties: {"variation":"control"}
    ````


- ### 🌕 Webview Viewed - origin property
  - #### 🟢 Feature flag ON
    ```
    17:59:46.436 WordPress-STATS      com.jetpack.android.beta   I  🔵 Tracked: domains_purchase_webview_viewed, Properties: {"origin":"menu","blog_id":215977563,"is_jetpack":false,"site_type":"blog"}
    ```

  - #### 🔴 Feature flag OFF
    The event still had the `origin` property when the feature flag is off for me:
    ```
    18:03:47.282 WordPress-STATS      com.jetpack.android.beta   I  🔵 Tracked: domains_purchase_webview_viewed, Properties: {"origin":"menu","blog_id":215977563,"is_jetpack":false,"site_type":"blog"}
    ```

## Other

> 5. Site created with paid domain event
> This could tracked in (1) but we should re-examine if a new property is needed in the site_created event

> I did not add the other 2 events mentioned in the issue as I am not sure how to test them also domains_registration_form_submitted seem to already exist in code. I think we need revisit them once the flow is in place.

Sounds good to me @antonis 👍

**Comment by antonis** [comment]
Thank you for reviewing and your feedback @ovitrif 🙇 

> I didn't see anything logic that depends on the feature flag in the changes though 🤷🏻 , which explains why during testing the new event properties are tracked regardless of the feature flag state (ON/OFF).

Good catch 👍 
I agree that the control variation is not supposed to change anything and implemented the change with 097a4e7bf98e79cf981fe4f218dface2d0d6c057

Ready for another pass 🤞

**Review by ovitrif — APPROVED** [review]
Everything LGMT after the last update @antonis 🙇🏻

Thank you for wrangling this and addressing my feedback 🥇 

I'll merge the PR 👍🏻 , any other events can be added when the functionality is available.

**Pull request merged**